libs/shared/l10n/src/lib/localizer-base.ts

```diff
@@ -0,0 +1,64 @@
+import { FluentBundle, FluentResource } from '@fluent/bundle';
+import { promises as fsPromises } from 'fs';
+
+export class LocalizerBase {
+  constructor(private ftlBasePath: string) {}
+
+  protected async fetchMessages(currentLocales: string[]) {
+    const fetchedPending: Record<string, Promise<string>> = {};
+    const fetched: Record<string, string> = {};
+    for (const locale of currentLocales) {
+      fetchedPending[locale] = this.fetchTranslatedMessages(locale);
+    }
+
+    // All we're doing here is taking `{ localeName: pendingLocaleMessagesPromise }` objects and
+    // parallelizing the promise resolutions instead of waiting for them to finish syncronously. We
+    // then return the result in the same `{ localeName: messages }` format for fulfilled promises.
+    const fetchedLocales = await Promise.allSettled(
+      Object.keys(fetchedPending).map(async (locale) => ({
+        locale,
+        fetchedLocale: await fetchedPending[locale],
+      }))
+    );
+
+    fetchedLocales.forEach((fetchedLocale) => {
+      if (fetchedLocale.status === 'fulfilled') {
```

Comment:
Q: Do we want to do anything (logging, etc) with any rejected promises?

Reply:
I've added a `console.error` in case of rejections. I don't think it'll hurt.